File: TabNine/Python/Q6.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        #Given a string s, find the length of the longest substring without repeating characters.
        
        if not s:
            return 0
        
        max_len = 0

        for i in range(len(s)):

            if s[i] not in s[i+1:]:
                max_len = max(max_len, i-i+1)

        return

File: Copilot/Java/Q6.java
package Java;

public class Q6 {
    public int lengthOfLongestSubstring(String s) {
        /*Given a string s, find the length of the longest substring
        *without repeating characters. 
        */
        int n = s.length();
        int ans = 0;
        int[] index = new int[128];
        for (int j = 0, i = 0; j < n; j++) {
            i = Math.max(index[s.charAt(j)], i);
            ans = Math.max(ans, j - i + 1);
            index[s.charAt(j)] = j + 1;
        }
        return ans;

    }
}
